### DISCUSSION TITLE
what does "local routing" config really affects, and in what situation it would be set to disabled(is_local_routing = false)?

### DISCUSSION THREAD

**Discussion by husher923** [COVESA/vsomeip#807] [Q&A] [answered]
been reading code of VSOMEIP for some time,its really help me a lot, thanks to the programmers.
my question is:
i see that if local routing is not configured,routing client would create a local_tcp_client_endpoint instead of a local_uds_client_endpoint.I want to know if a local_tcp_client_endpoint is set to communicate with nodes outside this device, and why in this case a netlink_connector is used 

I guess that netlink_connector is for monitoring  net device state change thus to inform the local_tcp_client_endpoint port/addr to connect to and local_tcp_client_endpoint would be able to receive SD related message directly without a rouing by routing_manager.

I'm not sure about above guess.

what so ever, again, thanks to those who would make VSOMEIP open to everyone.

**Comment by duartenfonseca** [comment, marked as answer]
if is_local_routing is set, vsomeip uses UDS for internal communications (it is the default). if not, then vsomeip uses TCP for internal communication.
local_tcp_client_endpoint only communicates internally in the device

**Reply by husher923** [reply]
Hi, thanks for your reply.Now I know the difference.
Since it seems a more complicated way to find a local endpoint by Netlink method, why need such a mechanism instead just using UDS endpoint, I'm wondering is that for different safety strategy?
